File: bubbly/builtin/schema_gen.go
package builtin

import "github.com/valocode/bubbly/api/core"

// #######################################
// _SCHEMA
// #######################################
type Schema struct {
	DBlock_Table  string                 `json:"_schema,omitempty"`
	DBlock_Policy core.DataBlockPolicy   `json:"-"`
	DBlock_Joins  []string               `json:"-"`
	Tables        map[string]interface{} `json:"tables,omitempty"`
}
type Schema_Wrap struct {
	Schema []Schema `json:"_schema,omitempty"`
}

// #######################################
// _RESOURCE
// #######################################
type Resource struct {
	DBlock_Table  string                 `json:"_resource,omitempty"`
	DBlock_Policy core.DataBlockPolicy   `json:"-"`
	DBlock_Joins  []string               `json:"-"`
	Id            string                 `json:"id,omitempty"`
	Name          string                 `json:"name,omitempty"`
	Kind          string                 `json:"kind,omitempty"`
	ApiVersion    string                 `json:"api_version,omitempty"`
	Spec          string                 `json:"spec,omitempty"`
	Metadata      map[string]interface{} `json:"metadata,omitempty"`
	Event         []Event                `json:"_event,omitempty"`
	ReleaseEntry  []ReleaseEntry         `json:"release_entry,omitempty"`
}
type Resource_Wrap struct {
	Resource []Resource `json:"_resource,omitempty"`
}

// #######################################
// _EVENT
// #######################################
type Event struct {
	DBlock_Table  string               `json:"_event,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Status        string               `json:"status,omitempty"`
	Error         string               `json:"error,omitempty"`
	Time          string               `json:"time,omitempty"`
	Resource      *Resource            `json:"_resource,omitempty"`
}
type Event_Wrap struct {
	Event []Event `json:"_event,omitempty"`
}

// #######################################
// RELEASE_ENTRY
// #######################################
type ReleaseEntry struct {
	DBlock_Table    string               `json:"release_entry,omitempty"`
	DBlock_Policy   core.DataBlockPolicy `json:"-"`
	DBlock_Joins    []string             `json:"-"`
	Name            string               `json:"name,omitempty"`
	Result          bool                 `json:"result,omitempty"`
	Reason          string               `json:"reason,omitempty"`
	Release         *Release             `json:"release,omitempty"`
	ReleaseCriteria *ReleaseCriteria     `json:"release_criteria,omitempty"`
	Resource        *Resource            `json:"_resource,omitempty"`
}
type ReleaseEntry_Wrap struct {
	ReleaseEntry []ReleaseEntry `json:"release_entry,omitempty"`
}

// #######################################
// RELEASE
// #######################################
type Release struct {
	DBlock_Table    string               `json:"release,omitempty"`
	DBlock_Policy   core.DataBlockPolicy `json:"-"`
	DBlock_Joins    []string             `json:"-"`
	Name            string               `json:"name,omitempty"`
	Version         string               `json:"version,omitempty"`
	Project         *Project             `json:"project,omitempty"`
	ReleaseItem     []ReleaseItem        `json:"release_item,omitempty"`
	ReleaseEntry    []ReleaseEntry       `json:"release_entry,omitempty"`
	ReleaseStage    []ReleaseStage       `json:"release_stage,omitempty"`
	ReleaseCriteria []ReleaseCriteria    `json:"release_criteria,omitempty"`
	CodeScan        []CodeScan           `json:"code_scan,omitempty"`
	TestRun         []TestRun            `json:"test_run,omitempty"`
}
type Release_Wrap struct {
	Release []Release `json:"release,omitempty"`
}

// #######################################
// PROJECT
// #######################################
type Project struct {
	DBlock_Table  string               `json:"project,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Name          string               `json:"name,omitempty"`
	Repo          []Repo               `json:"repo,omitempty"`
	Release       []Release            `json:"release,omitempty"`
}
type Project_Wrap struct {
	Project []Project `json:"project,omitempty"`
}

// #######################################
// REPO
// #######################################
type Repo struct {
	DBlock_Table  string               `json:"repo,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Id            string               `json:"id,omitempty"`
	Name          string               `json:"name,omitempty"`
	Project       *Project             `json:"project,omitempty"`
	Branch        []Branch             `json:"branch,omitempty"`
	Commit        []Commit             `json:"commit,omitempty"`
}
type Repo_Wrap struct {
	Repo []Repo `json:"repo,omitempty"`
}

// #######################################
// BRANCH
// #######################################
type Branch struct {
	DBlock_Table  string               `json:"branch,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Name          string               `json:"name,omitempty"`
	Repo          *Repo                `json:"repo,omitempty"`
	Commit        []Commit             `json:"commit,omitempty"`
}
type Branch_Wrap struct {
	Branch []Branch `json:"branch,omitempty"`
}

// #######################################
// COMMIT
// #######################################
type Commit struct {
	DBlock_Table  string               `json:"commit,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Id            string               `json:"id,omitempty"`
	Tag           string               `json:"tag,omitempty"`
	Time          string               `json:"time,omitempty"`
	Branch        *Branch              `json:"branch,omitempty"`
	Repo          *Repo                `json:"repo,omitempty"`
	ReleaseItem   *ReleaseItem         `json:"release_item,omitempty"`
}
type Commit_Wrap struct {
	Commit []Commit `json:"commit,omitempty"`
}

// #######################################
// RELEASE_ITEM
// #######################################
type ReleaseItem struct {
	DBlock_Table  string               `json:"release_item,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Type          string               `json:"type,omitempty"`
	Release       *Release             `json:"release,omitempty"`
	Commit        *Commit              `json:"commit,omitempty"`
	Artifact      *Artifact            `json:"artifact,omitempty"`
}
type ReleaseItem_Wrap struct {
	ReleaseItem []ReleaseItem `json:"release_item,omitempty"`
}

// #######################################
// ARTIFACT
// #######################################
type Artifact struct {
	DBlock_Table  string               `json:"artifact,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Name          string               `json:"name,omitempty"`
	Sha256        string               `json:"sha256,omitempty"`
	Location      string               `json:"location,omitempty"`
	ReleaseItem   *ReleaseItem         `json:"release_item,omitempty"`
}
type Artifact_Wrap struct {
	Artifact []Artifact `json:"artifact,omitempty"`
}

// #######################################
// RELEASE_STAGE
// #######################################
type ReleaseStage struct {
	DBlock_Table    string               `json:"release_stage,omitempty"`
	DBlock_Policy   core.DataBlockPolicy `json:"-"`
	DBlock_Joins    []string             `json:"-"`
	Name            string               `json:"name,omitempty"`
	Release         *Release             `json:"release,omitempty"`
	ReleaseCriteria []ReleaseCriteria    `json:"release_criteria,omitempty"`
}
type ReleaseStage_Wrap struct {
	ReleaseStage []ReleaseStage `json:"release_stage,omitempty"`
}

// #######################################
// RELEASE_CRITERIA
// #######################################
type ReleaseCriteria struct {
	DBlock_Table  string               `json:"release_criteria,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	EntryName     string               `json:"entry_name,omitempty"`
	ReleaseEntry  []ReleaseEntry       `json:"release_entry,omitempty"`
	ReleaseStage  *ReleaseStage        `json:"release_stage,omitempty"`
	Release       *Release             `json:"release,omitempty"`
}
type ReleaseCriteria_Wrap struct {
	ReleaseCriteria []ReleaseCriteria `json:"release_criteria,omitempty"`
}

// #######################################
// CODE_SCAN
// #######################################
type CodeScan struct {
	DBlock_Table  string               `json:"code_scan,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Tool          string               `json:"tool,omitempty"`
	Release       *Release             `json:"release,omitempty"`
	CodeIssue     []CodeIssue          `json:"code_issue,omitempty"`
}
type CodeScan_Wrap struct {
	CodeScan []CodeScan `json:"code_scan,omitempty"`
}

// #######################################
// CODE_ISSUE
// #######################################
type CodeIssue struct {
	DBlock_Table  string               `json:"code_issue,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Id            string               `json:"id,omitempty"`
	Message       string               `json:"message,omitempty"`
	Severity      string               `json:"severity,omitempty"`
	Type          string               `json:"type,omitempty"`
	CodeScan      *CodeScan            `json:"code_scan,omitempty"`
}
type CodeIssue_Wrap struct {
	CodeIssue []CodeIssue `json:"code_issue,omitempty"`
}

// #######################################
// TEST_RUN
// #######################################
type TestRun struct {
	DBlock_Table  string               `json:"test_run,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Tool          string               `json:"tool,omitempty"`
	Type          string               `json:"type,omitempty"`
	Name          string               `json:"name,omitempty"`
	Elapsed       int64                `json:"elapsed,omitempty"`
	Result        bool                 `json:"result,omitempty"`
	Release       *Release             `json:"release,omitempty"`
	TestCase      []TestCase           `json:"test_case,omitempty"`
}
type TestRun_Wrap struct {
	TestRun []TestRun `json:"test_run,omitempty"`
}

// #######################################
// TEST_CASE
// #######################################
type TestCase struct {
	DBlock_Table  string               `json:"test_case,omitempty"`
	DBlock_Policy core.DataBlockPolicy `json:"-"`
	DBlock_Joins  []string             `json:"-"`
	Name          string               `json:"name,omitempty"`
	Result        bool                 `json:"result,omitempty"`
	Message       string               `json:"message,omitempty"`
	TestRun       *TestRun             `json:"test_run,omitempty"`
}
type TestCase_Wrap struct {
	TestCase []TestCase `json:"test_case,omitempty"`
}

File: bubbly/builtin/schema_gen.ts
// #######################################
// _SCHEMA
// #######################################
export interface _schema {
	tables?: object;
}
export interface _schema_wrap {
	_schema?: _schema[];
}

// #######################################
// _RESOURCE
// #######################################
export interface _resource {
	id?: string;
	name?: string;
	kind?: string;
	api_version?: string;
	spec?: string;
	metadata?: object;
	_event?: _event[];
	release_entry?: release_entry[];
}
export interface _resource_wrap {
	_resource?: _resource[];
}

// #######################################
// _EVENT
// #######################################
export interface _event {
	status?: string;
	error?: string;
	time?: Date;
	_resource?: _resource;
}
export interface _event_wrap {
	_event?: _event[];
}

// #######################################
// RELEASE_ENTRY
// #######################################
export interface release_entry {
	name?: string;
	result?: boolean;
	reason?: string;
	release?: release;
	release_criteria?: release_criteria;
	_resource?: _resource;
}
export interface release_entry_wrap {
	release_entry?: release_entry[];
}

// #######################################
// RELEASE
// #######################################
export interface release {
	name?: string;
	version?: string;
	project?: project;
	release_item?: release_item[];
	release_entry?: release_entry[];
	release_stage?: release_stage[];
	release_criteria?: release_criteria[];
	code_scan?: code_scan[];
	test_run?: test_run[];
}
export interface release_wrap {
	release?: release[];
}

// #######################################
// RELEASE_CRITERIA
// #######################################
export interface release_criteria {
	entry_name?: string;
	release_stage?: release_stage;
	release?: release;
	release_entry?: release_entry[];
}
export interface release_criteria_wrap {
	release_criteria?: release_criteria[];
}

// #######################################
// RELEASE_STAGE
// #######################################
export interface release_stage {
	name?: string;
	release?: release;
	release_criteria?: release_criteria[];
}
export interface release_stage_wrap {
	release_stage?: release_stage[];
}

// #######################################
// CODE_SCAN
// #######################################
export interface code_scan {
	tool?: string;
	release?: release;
	lifecycle?: lifecycle;
	code_issue?: code_issue[];
}
export interface code_scan_wrap {
	code_scan?: code_scan[];
}

// #######################################
// LIFECYCLE
// #######################################
export interface lifecycle {
	status?: string;
	lifecycle_entry?: lifecycle_entry[];
	code_scan?: code_scan;
	test_run?: test_run;
}
export interface lifecycle_wrap {
	lifecycle?: lifecycle[];
}

// #######################################
// LIFECYCLE_ENTRY
// #######################################
export interface lifecycle_entry {
	author?: string;
	message?: string;
	signed?: boolean;
	lifecycle?: lifecycle;
}
export interface lifecycle_entry_wrap {
	lifecycle_entry?: lifecycle_entry[];
}

// #######################################
// TEST_RUN
// #######################################
export interface test_run {
	tool?: string;
	type?: string;
	name?: string;
	elapsed?: number;
	result?: boolean;
	test_case?: test_case[];
	release?: release;
	lifecycle?: lifecycle;
}
export interface test_run_wrap {
	test_run?: test_run[];
}

// #######################################
// TEST_CASE
// #######################################
export interface test_case {
	name?: string;
	result?: boolean;
	message?: string;
	test_run?: test_run;
}
export interface test_case_wrap {
	test_case?: test_case[];
}

// #######################################
// CODE_ISSUE
// #######################################
export interface code_issue {
	id?: string;
	message?: string;
	severity?: string;
	type?: string;
	code_scan?: code_scan;
}
export interface code_issue_wrap {
	code_issue?: code_issue[];
}

// #######################################
// PROJECT
// #######################################
export interface project {
	name?: string;
	repo?: repo[];
	release?: release[];
}
export interface project_wrap {
	project?: project[];
}

// #######################################
// REPO
// #######################################
export interface repo {
	id?: string;
	name?: string;
	branch?: branch[];
	commit?: commit[];
	project?: project;
}
export interface repo_wrap {
	repo?: repo[];
}

// #######################################
// BRANCH
// #######################################
export interface branch {
	name?: string;
	commit?: commit[];
	repo?: repo;
}
export interface branch_wrap {
	branch?: branch[];
}

// #######################################
// COMMIT
// #######################################
export interface commit {
	id?: string;
	tag?: string;
	time?: string;
	branch?: branch;
	repo?: repo;
	release_item?: release_item;
}
export interface commit_wrap {
	commit?: commit[];
}

// #######################################
// RELEASE_ITEM
// #######################################
export interface release_item {
	type?: string;
	commit?: commit;
	artifact?: artifact;
	release?: release;
}
export interface release_item_wrap {
	release_item?: release_item[];
}

// #######################################
// ARTIFACT
// #######################################
export interface artifact {
	name?: string;
	sha256?: string;
	location?: string;
	release_item?: release_item;
}
export interface artifact_wrap {
	artifact?: artifact[];
}

